Training/Training/file.c

```diff
@@ -7,7 +7,135 @@
 // ------------------------------------------------------------------------------------------------
 
 #include <stdio.h>
+#include<stdbool.h>
+#include <string.h>
+#include <ctype.h>
+#include <stdlib.h>
+#define MAX 13
+
+int GetUserInput (char* prompt) {
+   int number = 0;
+   if (prompt != "") printf ("%s", prompt);
+   for (;;) {
+      char strNum[MAX], * endPtr = NULL,
+         * strPtr = fgets (strNum, sizeof (strNum), stdin),
+         * validInput = strchr (strNum, '\n');
+      number = strtol (strNum, &endPtr, 10);
+      if (validInput == NULL)  while (getchar () != '\n');
+      if (strPtr == NULL || (!isdigit (strNum[0]) && strNum[0] != '-') || *endPtr != '\n') {
+         printf ("Invalid Number!!! Please re-enter\n");
+         continue;
+      }
+      break;
+   }
+   return number;
+}
+
+
+void PrintBoard (char pieces[]) {
+   printf ("BOARD GAME\n");
+   printf ("+---+---+---+\n|");
+   for (int j = 0; j < 9; j++) {
+      switch (j) {
+         case 0:
+            printf (" %c |", pieces[j]);
+            break;
+         case 1:
+            printf (" %c |", pieces[j]);
+            break;
+         case 2:
+            printf (" %c |", pieces[j]);
+            break;
+         case 3:
+            printf ("| %c |", pieces[j]);
+            break;
+         case 4:
+            printf (" %c |", pieces[j]);
+            break;
+         case 5:
+            printf (" %c |", pieces[j]);
+            break;
+         case 6:
+            printf ("| %c |", pieces[j]);
+            break;
+         case 7:
+            printf (" %c |", pieces[j]);
+            break;
+         case 8:
+            printf (" %c |", pieces[j]);
+            break;
+         default:
+            break;
+      }
+      if ((j + 1) % 3 == 0) {
+         printf ("\n");
+         printf ("+---+---+---+\n");
+      }
+   }
+}
+
+void ClrScreen () {
+#ifdef _WIN32
+   system ("cls");
+#endif 
+}
 
 void main () {
-	printf ("Hello World");
+   char pieces[] = { ' ',' ',' ',' ',' ',' ',' ',' ',' ' };
+   PrintBoard (pieces);
+   int player1 = 0;
+   int player2 = 0;
+   int input1 = 0, input2 = 0;
+   for (int i = 0; i < 4; i++) {
+      for (;;) {
+         input1 = GetUserInput ("Player1:");
```

Comment:
You should display the board after each player makes a move and also check for win/draw after each move and not after both players move.